### ISSUE TITLE
Field set to Not visible still visible when displayed

### ISSUE THREAD

**Issue by Kemmotar83** [PoetOS/moodle-local_metadata#11]
Hi,

I've created a set of metadata in course context and then I'm using the _local_metadata_display_fields_ (lib.php) function to display them.
I've set a field _Not visible_ but, when I display the fields with above function, it's still visible, even if I view the output as a guest user.

I've checked the code and the _local_metadata_display_fields_ (lib.php) function calls _$formfield->is_visible()_ (raw 163) in order to check if the field should be displayed or not and in this function it's never checked if the field is set to "Not visible" (PROFILE_VISIBLE_NONE), that's why it is always displayed. Adding the PROFILE_VISIBLE_NONE case solve the "problem".
Am I missing something?

Other than that, in order to use the _local_metadata_display_fields_ function I've included the lib.php file but, when the code uses the PROFILE_VISIBLE_* constants, it seems they are not defined so they cannot be correctly evaluated. Do I need to include anything else in order to have these constants defined?
Could be better to define local constant, instead of using USER constants?

Thanks for this project. Moodle definitely needs this feature.

Giorgio

**Comment by mchurchward** [comment]
I guess the question is, what capability is relevant in making course metadata visible/invisible? I don't think it should be the same one used for user profile data. Or should new capabilities be created per context?

**Comment by Kemmotar83** [comment]
Yes, maybe a single capability _View hidden metadata_ per context could be the best solution.
Then, for each metadata instance, the visibility setting could just be _Visible_ or _Hidden_.

Thanks,
Giorgio
